validator/enforce_labels_test.rego

```diff
@@ -0,0 +1,201 @@
+	violations[_].details.resource == "//storage.googleapis.com/bucket-with-label1-missing"
+}
+
+#### Testing for GCE resources
+
+#### Testing for GCE instances
+# # Confirm exactly 4 instance violations were found
+test_enforce_label_compute_instance_violates_project_four {
+	violations := compute_instance_violations
+	count(violations) == 4
+	violation := violations[_]
+	is_string(violation.msg)
+	is_object(violation.details)
+}
+
+# confirm which 3 instances are in violation
+test_enforce_label_compute_instance_violates_project_basic {
+	violations := compute_instance_violations
+	violations[_].details.resource == "//compute.googleapis.com/projects/vpc-sc-pub-sub-billing-alerts/zones/us-central1-b/instances/invalid-instance-missing-labels-8hz5"
+	violations[_].details.resource == "//compute.googleapis.com/projects/vpc-sc-pub-sub-billing-alerts/zones/us-central1-b/instances/invalid-instance-missing-label1-8hz5"
+	violations[_].details.resource == "//compute.googleapis.com/projects/vpc-sc-pub-sub-billing-alerts/zones/us-central1-b/instances/invalid-instance-missing-label2-8hz5"
+}
+
+#### Testing for GCE Images
+# # Confirm exactly 4 images violations were found
+test_enforce_label_compute_instance_violates_project_four {
+	violations := compute_image_violations
+	count(violations) == 4
+	violation := violations[_]
+	is_string(violation.msg)
+	is_object(violation.details)
+}
+
+# confirm which 3 images are in violation
+test_enforce_label_compute_image_violates_project_basic {
+	violations := compute_image_violations
+	violations[_].details.resource == "//compute.googleapis.com/projects/my-own-project/global/images/test-invalid-image-missing-label1"
+	violations[_].details.resource == "//compute.googleapis.com/projects/my-own-project/global/images/test-invalid-image-missing-label2"
+	violations[_].details.resource == "//compute.googleapis.com/projects/my-own-project/global/images/test-invalid-image-missing-label1"
+}
+
+#### Testing for GCE Disks
+# # Confirm exactly 4 disk violations were found
+test_enforce_label_compute_disk_violates_project_four {
+	violations := compute_disk_violations
+	count(violations) == 4
+	violation := violations[_]
+	is_string(violation.msg)
+	is_object(violation.details)
+}
+
+# confirm which 3 disks are in violation
+test_enforce_label_compute_disk_violates_project_basic {
+	violations := compute_disk_violations
+
+	violations[_].details.resource == "//compute.googleapis.com/projects/my-test-project/zones/us-east1-b/disks/instance-1-invalid-disk-missing-labels"
+	violations[_].details.resource == "//compute.googleapis.com/projects/my-test-project/zones/us-east1-b/disks/instance-1-invalid-disk-missing-label1"
+	violations[_].details.resource == "//compute.googleapis.com/projects/my-test-project/zones/us-east1-b/disks/instance-1-invalid-disk-missing-label2"
+}
+
+#### Testing for GCE Snapshots
+# # Confirm exactly 4 snapshot violations were found
+test_enforce_label_compute_snapshot_violates_project_four {
```

Comment:
This name seems off. Why do we include `project` in all of them?

```suggestion
test_enforce_label_compute_snapshot_violates_project_four {
```

Reply:
true, the name is wrong, will fix